src/main/java/io/github/mirromutth/r2dbc/mysql/MySqlConnectionFactory.java

```diff
@@ -53,19 +52,15 @@ public static MySqlConnectionFactory from(MySqlConnectionConfiguration configura
         requireNonNull(configuration, "configuration must not be null");
 
         return new MySqlConnectionFactory(Mono.defer(() -> {
-            MySqlSession session = new MySqlSession(
-                configuration.getDatabase(),
-                configuration.getZeroDateOption(),
-                configuration.getUsername(),
-                configuration.getPassword()
-            );
+            MySqlSession session = new MySqlSession(configuration.getDatabase(), configuration.getZeroDateOption());
             String host = configuration.getHost();
             int port = configuration.getPort();
             Duration connectTimeout = configuration.getConnectTimeout();
             MySqlSslConfiguration sslConfiguration = configuration.getSslConfiguration();
+            Boolean requireSsl = sslConfiguration == null ? null : false;
```

Comment:
SSL requires should be considered by another PR, this PR considers only SSL supported by server or not.